File: app/src/main/java/com/byoutline/kickmaterial/dagger/GlobalComponent.java
package com.byoutline.kickmaterial.dagger;

import com.byoutline.kickmaterial.KickMaterialApp;
import com.byoutline.kickmaterial.activities.CategoriesListActivity;
import com.byoutline.kickmaterial.activities.ProjectDetailsActivity;
import com.byoutline.kickmaterial.activities.RewardsListActivity;
import com.byoutline.kickmaterial.fragments.ProjectsListFragment;
import com.byoutline.kickmaterial.fragments.SearchListFragment;
import com.byoutline.kickmaterial.managers.AccessTokenProvider;
import com.squareup.otto.Bus;
import dagger.Component;

/**
 * Created by Sebastian Kacprzak <sebastian.kacprzak at byoutline.com> on 27.03.15.
 */
@GlobalScope
@Component(modules = GlobalModule.class)
public interface GlobalComponent {
    void inject(CategoriesListActivity fragment);

    void inject(SearchListFragment fragment);

    void inject(ProjectsListFragment fragment);

    void inject(ProjectDetailsActivity activity);

    void inject(RewardsListActivity rewardsListActivity);

    Bus getBus();

    KickMaterialApp getApp();

    AccessTokenProvider getAccessTokenProvider();
}

File: app/src/main/java/com/byoutline/kickmaterial/dagger/GlobalComponent.kt
package com.byoutline.kickmaterial.dagger

import com.byoutline.kickmaterial.KickMaterialApp
import com.byoutline.kickmaterial.features.projectdetails.ProjectDetailsActivity
import com.byoutline.kickmaterial.features.projectlist.ProjectsListFragment
import com.byoutline.kickmaterial.features.rewardlist.RewardsListActivity
import com.byoutline.kickmaterial.features.search.SearchListFragment
import com.byoutline.kickmaterial.features.selectcategory.CategoriesListActivity
import com.byoutline.kickmaterial.model.DiscoverQuery
import com.byoutline.kickmaterial.model.DiscoverResponse
import com.byoutline.observablecachedfield.ObservableCachedFieldWithArg
import dagger.Component

/**
 * Created by Sebastian Kacprzak <sebastian.kacprzak at byoutline.com> on 27.03.15.
 */
@GlobalScope
@Component(modules = arrayOf(GlobalModule::class))
interface GlobalComponent {
    fun inject(fragment: CategoriesListActivity)
    fun inject(fragment: SearchListFragment)
    fun inject(fragment: ProjectsListFragment)
    fun inject(activity: ProjectDetailsActivity)
    fun inject(rewardsListActivity: RewardsListActivity)

    val app: KickMaterialApp
    val discoverField: ObservableCachedFieldWithArg<DiscoverResponse, DiscoverQuery>
}
